src/erlcloud_application_autoscaler.erl

```diff
@@ -0,0 +1,689 @@
+                               ResourceId :: binary(),
+                               ScalableDimension :: binary(),
+                               ServiceNamespace :: binary()) -> response().
+register_scalable_target(Configuration, ResourceId, ScalableDimension, ServiceNamespace) ->
+    BodyProps = [{<<"ResourceId">>, ResourceId},
+                 {<<"ScalableDimension">>, ScalableDimension},
+                 {<<"ServiceNamespace">>, ServiceNamespace}],
+    register_scalable_target(Configuration, BodyProps).
+
+-spec register_scalable_target(Configuration :: aws_config(),
+                               ResourceId :: binary(),
+                               ScalableDimension :: binary(),
+                               ServiceNamespace :: binary(),
+                               ResourceARN :: binary()) -> response().
+register_scalable_target(Configuration, ResourceId, ScalableDimension, ServiceNamespace, ResourceARN) ->
+    BodyProps = [{<<"ResourceId">>, ResourceId},
+                 {<<"ScalableDimension">>, ScalableDimension},
+                 {<<"ServiceNamespace">>, ServiceNamespace},
+                 {<<"ResourceARN">>, ResourceARN}],
+    register_scalable_target(Configuration, BodyProps).
+
+-spec register_scalable_target(Configuration :: aws_config(),
+                               ResourceId :: binary(),
+                               ScalableDimension :: binary(),
+                               ServiceNamespace :: binary(),
+                               ResourceARN :: binary(),
+                               MinCapacity :: integer() | undefined,
+                               MaxCapacity :: integer() | undefined) -> response().
+register_scalable_target(Configuration, ResourceId, ScalableDimension, ServiceNamespace, ResourceARN, MinCapacity, MaxCapacity) ->
+    BodyProps = [{<<"ResourceId">>, ResourceId},
+                 {<<"ScalableDimension">>, ScalableDimension},
+                 {<<"ServiceNamespace">>, ServiceNamespace},
+                 {<<"ResourceARN">>, ResourceARN}],
+    MaybeBodyWithMax = case MaxCapacity of
+        undefined -> [];
+        Max -> [{<<"MaxCapacity">>, Max}]
+    end,
+    MaybeBodyWithMin = case MinCapacity of
+        undefined -> [];
+        Min -> [{<<"MinCapacity">>, Min}]
+    end,
+    register_scalable_target(Configuration, BodyProps ++ MaybeBodyWithMax ++ MaybeBodyWithMin).
+
+-spec register_scalable_target(
+    Configurations :: erlcloud_aws:aws_config(),
+    BodyConfigurations :: aws_aas_request_body()
+) -> response().
+register_scalable_target(Configurations, BodyConfigurations) ->
+    request_with_action(Configurations, BodyConfigurations, "AnyScaleFrontendService.RegisterScalableTarget").
+
+
+%%%------------------------------------------------------------------------------
+%%% Internal functions.
+%%%------------------------------------------------------------------------------
+
+request_with_action(Configuration, BodyConfigurations, Action) ->
```

Comment:
why not using already erlcloud_retry module?

Reply:
I was taking as a basis the `erlcloud_ddb2` module and the implementation of its requests (`erlcloud_ddb_impl`), so I didn't acknowledge a possible implementation using `erlcloud_retry`.